src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -397,28 +397,58 @@ Override targets which don't work with project files that are not present on dis
         static string EscapeValue(string value) => SecurityElement.Escape(value);
     }
 
+#pragma warning disable RSEXPERIMENTAL003 // 'SyntaxTokenParser' is experimental
+#pragma warning disable RSEXPERIMENTAL005 // 'IgnoredDirectiveTriviaSyntax' is experimental
     public static ImmutableArray<CSharpDirective> FindDirectives(SourceFile sourceFile)
     {
         var builder = ImmutableArray.CreateBuilder<CSharpDirective>();
+        SyntaxTokenParser tokenizer = SyntaxFactory.CreateTokenParser(sourceFile.Text,
+            CSharpParseOptions.Default.WithFeatures([new("FileBasedProgram", "true")]));
 
-        // NOTE: When Roslyn is updated to support "ignored directives", we should use its SyntaxTokenParser instead.
-        foreach (var line in sourceFile.Text.Lines)
+        var result = tokenizer.ParseLeadingTrivia();
+        TextSpan previousWhiteSpaceSpan = default;
+        foreach (var trivia in result.Token.LeadingTrivia)
         {
-            var lineText = sourceFile.Text.ToString(line.Span);
+            // Stop when the trivia contains an error (e.g., because it's after #if).
```

Comment:
I wonder if we should be more specific about what kinds of errors make us stop. Essentially if something happens to be malformed is doing that likely to prevent a cascade or produce a result which is more actionable. Perhaps we can simply adjust if we find that to be the case.

Reply:
During 'dotnet run file.cs', compiler reports errors that I think are actionable. For 'dotnet project convert', addressing your other comment should start producing more actionable errors there as well.